### DISCUSSION TITLE
Rationale behind passing block to ComponentManager.addType

### DISCUSSION THREAD

**Discussion by membla** [GrapesJS/grapesjs#6214] [Q&A]
I need to keep track of the relation between components and blocks and had done so using a custom property on the root component of my blocks. However, I just now saw there is a `block` parameter on `ComponentManager.addType` that after a quick glance looks like a shortcut to add a Block at the same time we add a Component.

Couldn't find any mentions of this in the docs so I'm wondering what the general idea behind the `block` parameter is, to allow me to decide whether I should just scrap my custom solution. Is it simply a way to make the creation of component/block pairs less verbose or is it meant to support tracking the relationship between components and blocks?

**Comment by artf** [comment]
> Is it simply a way to make the creation of component/block pairs less verbose

This was my reason 😅

**Reply by membla** [reply]
Thank you! I reckon I'll keep my custom solution to tracking component/blocks relationships then.
